src/oracle.c

```diff
@@ -592,6 +599,80 @@ static int o_read_database_query(o_database_t *db, /* {{{ */
 #undef ALLOC_OR_FAIL
 } /* }}} int o_read_database_query */
 
+static int o_read_password_command(o_database_t *db) /* {{{ */
+{
+  const size_t pw_increment = 128;
+  char cmdbuf[4096], *sp, *bp, *ep = cmdbuf + sizeof cmdbuf, *fmtval;
+  char *pass, *nl;
+  size_t remain, w, r, px;
+  FILE *out;
+
+  for (sp = db->password_cmd, bp = cmdbuf; bp < ep;) {
+    switch (*sp) {
+    case '\0':
+      *bp++ = '\0';
+      goto success;
+    case '%':
```

Comment:
I'd prefer if you would use `subst_string()` from `"utils_subst.h"` to replace the placeholders `%{username}` and `%{connect_id}`.